src/components/More.js

```diff
@@ -1,18 +1,45 @@
 import React from "react";
 import Radium from "radium";
 import Button from "./Button";
+import {isMoreMenuExpanded} from "../redux/selectors";
+import {expandMoreMenu, collapseMoreMenu} from '../redux/actionCreators';
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import {styles} from './styles';
 
-//TODO
-//implement correct behavior
 export class More extends React.Component {
   render() {
+    if (this.props.expanded) {
+      return(
+        <Button kind="toolbar"
+                style={styles.buttons.more.moreButton}
+                onClick={()=> this.props.collapse()}>
+          More
+        </Button>
+      );
+    }
     return(
       <Button kind="toolbar"
-              onClick={()=> this.setState({expanded: true})}>
+              style={styles.buttons.more.moreButton}
+              onClick={()=> this.props.expand()}>
```

Comment:
the only thing that is different between these two conditions is what the onClick handler does. I would just put the condition in there. i.e:

```jsx
return (
  <Button kind="toolbar"
          style={styles.buttons.more.moreButton}
          onClick={() => this.state.expanded ? this.props.collapse() : this.props.expand()}>
    More
  </Button>
);
```